src/renderer/widgets/WalletDetails/model/__tests__/rename-wallet-model.test.ts

```diff
@@ -0,0 +1,44 @@
+import { fork, allSettled } from 'effector';
+
+import { walletMock } from './mocks/wallet-mock';
+import { renameWalletModel } from '../rename-wallet-model';
+import { walletModel } from '@entities/wallet';
+import { walletSelectModel } from '@features/wallets';
+
+jest.mock('@app/providers', () => ({
+  useMatrix: jest.fn(),
+}));
+
+describe('entities/wallet/model/wallet-model', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('should discard not unique wallet names', async () => {
+    const wallets = [walletMock.wallet1, walletMock.wallet2];
+    const scope = fork({
+      values: new Map().set(walletModel.$wallets, wallets),
+    });
+
+    await allSettled(renameWalletModel.events.formInitiated, { scope, params: walletMock.wallet1 });
+    await allSettled(renameWalletModel.$walletForm.setForm, { scope, params: { name: walletMock.wallet2.name } });
+    await allSettled(renameWalletModel.$walletForm.validate, { scope });
+
+    expect(scope.getState(renameWalletModel.$walletForm.$isValid)).toEqual(false);
+  });
+
+  test('should updated $walletForDetails and $wallets after name change', async () => {
```

Comment:
Tests must cover only `wallet-rename-model` data flow
I'll ask in chat how to test data flow after form submission